### ISSUE TITLE
lang attribute: validate BCP 47 subtags against the IANA registry, not just syntax

### ISSUE THREAD

**Issue by YusukeHirao** [markuplint/markuplint#3963]
## Summary

`isBCP47` (`packages/@markuplint/types/src/rfc/is-bcp-47.ts`) validates the `lang` attribute's syntactic *shape* only, via the `bcp-47` npm package's `parse()`. It never checks whether the individual subtags (primary language, extlang, script, region, variant) are actually registered in the IANA Language Subtag Registry, so tags with well-formed-but-nonexistent subtags pass silently.

## Repro

```html
<html lang="zzz">
```
```html
<body lang="bat-smg">
```

nu-validator (bench reference):
- `zzz`: error — `Bad value "zzz" for attribute "lang" on element "html": The language subtag "zzz" is not a valid ISO language part of a language tag.`
- `bat-smg`: error — `Bad value "bat-smg" for attribute "lang" on element "body": Bad extlang subtag "smg".`

markuplint (all bench rules enabled): 0 violations for both.

Fixtures (`nu-only` in `tests/external/snapshots/diff/nu-only.json`):
- `tests/external/validator/tests/html/attributes/lang/invalid-primary-novalid.html`
- `tests/external/validator/tests/html/attributes/lang/extlang-bad-novalid.html`

## Spec basis

[HTML LS §3.2.6.2 "The lang and xml:lang attributes"](https://html.spec.whatwg.org/multipage/dom.html#attr-lang):

> Its value must be a valid BCP 47 language tag, or the empty string. [...] [BCP47]

BCP 47 (RFC 5646) §2.2.2 defines the `extlang` and primary `language` subtags as registered against the IANA Language Subtag Registry — a syntactically well-formed 2-3 letter subtag is not automatically a *valid* one; it must appear in the registry.

## Current implementation

`packages/@markuplint/types/src/rfc/is-bcp-47.ts`:

```ts
export const isBCP47: FormattedPrimitiveTypeCreator = () => {
	return value => {
		const { language, privateuse } = parse(value);
		return !!language || (privateuse != null && privateuse.length > 0);
	};
};
```

`bcp-47`'s `parse()` only checks the ABNF grammar shape (does a `language` subtag exist syntactically, is `privateuse` present). It has no registry data, so `zzz` and `bat-smg` both parse successfully.

## Proposed direction

Replace or augment the `bcp-47`-based check with a registry-aware validator. Candidate: [`language-tags`](https://www.npmjs.com/package/language-tags) (npm, v2.1.0 as of 2025-05-13, MIT license) — bundles IANA Language Subtag Registry data and exposes `check(tag)` / `tags(tag).valid()` plus `tag.errors()` for diagnostics. This would let the rule distinguish "malformed BCP 47 tag" from "well-formed but references a non-registered subtag" and potentially surface the specific bad subtag in the violation message the way nu-validator does.

Scope note: the registry dataset needs a periodic-refresh story (the registry itself is updated by IANA over time) — worth deciding whether to vendor a snapshot or depend on a package that publishes updates.

## Files

- `packages/@markuplint/types/src/rfc/is-bcp-47.ts`
- `packages/@markuplint/types/src/rfc/is-bcp-47.spec.ts`
- `packages/@markuplint/types/package.json` (new dependency)

## Bench impact

2 fixtures currently `nu-only`. Resolving this Issue is expected to flip both to `match-error`.

**Comment by YusukeHirao** [comment]
Duplicate of #3829 — closing in favor of the existing, more thorough Issue. Added the `language-tags` package suggestion as a comment there.
